### ISSUE TITLE
Build error - Type 'string' does not satisfy the constraint 'object'.

### ISSUE THREAD

**Issue by asafron** [angular-redux/form#14]
When trying to add it to my project using:

      npm install @angular-redux/form

I'm getting the following error:

Asafs-MBP:MyProject asafron$ npm install @angular-redux/form

> @angular-redux/form@6.0.1 postinstall /Documents/MyProject/node_modules/@angular-redux/form
> npm run build

> @angular-redux/form@6.0.1 prebuild /Documents/MyProject/node_modules/@angular-redux/form
> npm run clean

> @angular-redux/form@6.0.1 clean /Documents/MyProject/node_modules/@angular-redux/form
> rimraf dist

> @angular-redux/form@6.0.1 build /Documents/MyProject/node_modules/@angular-redux/form
> ngc

Error at /Documents/MyProject/node_modules/@angular-redux/form/source/state.ts:221:25: Type 'string' does not satisfy the constraint 'object'.

npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v6.9.1
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE
npm ERR! @angular-redux/form@6.0.1 build: `ngc`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @angular-redux/form@6.0.1 build script 'ngc'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @angular-redux/form package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ngc
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @angular-redux/form
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @angular-redux/form
npm ERR! There is likely additional logging output above.

**Comment by Cibico99** [comment]
+1
Here is my npm-debug.log
[npm-debug.log.txt](https://github.com/angular-redux/form/files/821535/npm-debug.log.txt)

**Comment by Sieabah** [comment]
+1

It's a blocker to even install it

**Comment by javico2609** [comment]
npm ERR! @angular-redux/form@6.0.1 postinstall: `npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @angular-redux/form@6.0.1 postinstall script 'npm run build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @angular-redux/form package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @angular-redux/form
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @angular-redux/form
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

**Comment by serhiiharbo** [comment]
@javico2609 @Sieabah @Cibico99 @asafron - hope it will be still useful for you!
I found a solution how to install @angular-redux/form to my project: **try to decrease you TypeScript version to 2.1.6 in package.json** instead of all you have above this version.

Here is explanation how I explore this solution: I have two identical angular2 projects, based on the single generator. So for the first one @angular-redux/form installs perfectly, here it is [angular2-redux](https://github.com/aquile/angular2-redux). And for another one it installs with this error _"Error at /Documents/MyProject/node_modules/@angular-redux/form/source/state.ts:221:25: Type 'string' does not satisfy the constraint 'object'."_ yet  I have found the only one cruel difference between them - it is TypeScript lang version. Project, where I couldn't install @angular-redux/form TypeScript version was ^2.2.1 vs ^2.1.6 in project with no issues with @angular-redux/form installation.
Hope you doesn't waste your time reading that))) Good luck!
